libbindgen/src/ir/ty.rs

```diff
@@ -554,7 +554,7 @@ impl Type {
                     TypeKind::Function(signature)
                     // Same here, with template specialisations we can safely
                     // assume this is a Comp(..)
-                } else if ty.template_args().map_or(false, |x| x.len() > 0) {
+                } else if ty.is_fully_specialized_template() {
```

Comment:
Is there a test case for these changes? I'm not really sure what their purpose is... Or that its more correct than before? (Not that I don't believe you or anything ;) Just that this commit doesn't really stand on its own, as written.)